### PR TITLE
KAA-1583: REST log appender - reviewed. [master]

### PR DESCRIPTION
**Pull request by hunch7**

### CODE REVIEW

File: doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md [hunk 1]
```diff
@@ -2,94 +2,42 @@
...
 {:toc}
 
-The REST log appender is responsible for transferring logs from Operations service to your custom service.
+The REST log appender is used to transfer logs from the [Operations service]({{root_url}}Glossary/#operations-service) to your custom service.
 
-# Creating REST log appender with Admin UI
+## Create REST log appender
 
-The easiest way to create REST log appender for your application is by using Admin UI.
+To create a Cassandra log appender for your application using the [Administration UI]({{root_url}}Glossary/#administration-ui):
```
Comment (Polinaa) [inline doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md]: REST log appender

File: doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md [hunk 2]
```diff
@@ -142,114 +90,156 @@ Example result:
...
+...
+private KaaClient client;
+client = ... ;
+
+Data data = new Data(...);
+client.addLogRecord(data);
+...
+```
+
+After sending logs from [Kaa client]({{root_url}}Glossary/kaa-client), [Kaa server]({{root_url}}Glossary/kaa-server) will use previously created REST log appender which will send data to your custom service based on above code.
```
Comment (Polinaa) [inline doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md]: Broken links: _Kaa client_,  _Kaa server_.

File: doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md [hunk 3]
```diff
@@ -142,114 +90,156 @@ Example result:
...
+1. Log in to the **Administration UI** as a tenant administrator and create an application that uses a trustful credentials service.
+	![Enter application title](attach/add-application2.png)
+
+    >**NOTE:** To see the [Application token]({{root_url}}Glossary/#application-token) generated for your new application, open the **Application details** window containing the application token.<br/>
+
+2.  Log in to the **Administration UI** as a tenant developer and create a log schema for your new application.
+To do this, select your application in the list and click **Schemas** > **Log** > **Add schema**.
+	![Add log schema](attach/rest-log-appender1.png)
+
+3. Upload the following configuration schema.
```
Comment (Polinaa) [inline doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md]: This step refers to the previous one.

File: doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md [hunk 4]
```diff
@@ -142,114 +90,156 @@ Example result:
...
+
+		{
```
Comment (Polinaa) [inline doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md:150]: Can't upload the schema. Used the following schema:
```
{
             "type":"record",
             "name":"Data",
             "namespace":"org.kaaproject.kaa.example.mobile.log",
             "fields":[
                 {
                     "name":"timestamp",
                     "type":[
                         "long",
                         "null"
                     ]
                 },
                 {
                     "name":"data",
                     "type":[
                         "bytes",
                         "null"
                     ]
                 },
                 {
                     "name":"endpointKeyHash",
                     "type":[
                         {
                             "type":"string",
                             "avro.java.string":"String"
                         },
                         "null"
                     ]
                 },
                 {
                     "name":"hashFunction",
                     "type":[
                         {
                             "type":"string",
                             "avro.java.string":"String"
                         },
                         "null"
                     ]
                 }
             ]
}
```

File: doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md [hunk 5]
```diff
@@ -142,114 +90,156 @@ Example result:
...
+		}
+
+4. Add your custom log appender to your new application.
+To do this, open the **Log appenders** page of your application and click **Add log appender**.
+Set up your log appender configuration and click **Add**.
+	![Add log appender](attach/rest-log-appender2.png)
+
+5. Generate an [SDK for your platform]({{root_url}}Programming-guide/Using-Kaa-endpoint-SDKs/) and add the downloaded SDK to your project directory.
+
+6. Run a sample application from the [Kaa Sandbox]({{root_url}}Glossary/kaa-sandbox) to test your new log appender.
```
Comment (Polinaa) [inline doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md]: Please delete this step, user doesn't use any demo from Sandbox in this guide.

File: doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md (hunk 1 above)
Comment (hunch7) [reply doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md]: Fixed.

File: doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md (hunk 2 above)
Comment (hunch7) [reply doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md]: Fixed.

File: doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md (hunk 3 above)
Comment (hunch7) [reply doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md]: Fixed.

File: doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md (hunk 4 above)
Comment (hunch7) [reply doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md:150]: Fixed.

File: doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md (hunk 5 above)
Comment (hunch7) [reply doc/Programming-guide/Key-platform-features/Data-collection/Rest-log-appender/index.md]: Fixed.

**Review by Polinaa — APPROVED** [review]

**Review by vadimol — APPROVED** [review]
Backport for #1408

**Pull request merged**